### PR TITLE
feat(pagination): CSS-in-JS refactor

### PR DESCRIPTION
**Pull request by jzempel**
- [x] **BREAKING CHANGE?** 

## Description

All CSS moved into themed JS, making `styles.css` import unnecessary. Bundle size reduction from apx 19kb (12kb JS + 7kb CSS) down to 7.75kb combined. :tada:

Fixes:
- Pagination `renderPages` logic corrected to properly handle pages, padding, and gaps.
- Updated with improved ARIA for accessibility.

Feature enhancements:
- New `pageGap` prop for specifying gap positioning

## Detail

Demo pre-published for review: https://competent-shirley-cc2f82.netlify.com/pagination/

## Checklist

- [x] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
- [ ] :nail_care: ~view component styling is based on a Garden CSS
      component~
- [x] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [x] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [x] :guardsman: includes new unit tests
- [x] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: packages/pagination/src/elements/Pagination.spec.tsx [hunk 1]
```diff
@@ -8,10 +8,16 @@
 import React from 'react';
 import { render, fireEvent } from 'garden-test-utils';
 import { KEY_CODES } from '@zendeskgarden/container-utilities';
-
-import Pagination, { IPaginationProps, PAGE_TYPE } from './Pagination';
+import { default as Pagination, IPaginationProps, PAGE_TYPE } from './Pagination';
```
Comment (austin94) [inline packages/pagination/src/elements/Pagination.spec.tsx]: Should we use the standard default export? i.e.

```ts
import Pagination, { IPaginationProps, PAGE_TYPE } from '...';
```

File: packages/pagination/src/elements/Pagination.tsx [hunk 2]
```diff
@@ -35,18 +31,27 @@ export interface IPaginationProps extends Omit<HTMLAttributes<HTMLUListElement>,
...
   /**
-   * @param {Any} currentPage - The newly selected page
+   * Position for the leading and trailing gap indicator, if needed based on
+   * current and total pages
    */
-  onChange?: (updatedCurrentPage: number) => void;
+  pageGap?: number;
   /**
-   * Allows custom props to be applied to each page element. Useful for QA attributes and localization.
+   * @param {Any} currentPage - The currently selected page
```
Comment (austin94) [inline packages/pagination/src/elements/Pagination.tsx]: These JSDoc values are already created by TSDoc in the prop-sheet. Do we need to duplicate these here?

File: packages/pagination/src/elements/Pagination.spec.tsx (hunk 1 above)
Comment (jzempel) [reply packages/pagination/src/elements/Pagination.spec.tsx]: No, without `default as ...` we miss the ability to test the `ref` exported via the default `withTheme` export.

File: packages/pagination/src/elements/Pagination.tsx (hunk 2 above)
Comment (jzempel) [reply packages/pagination/src/elements/Pagination.tsx]: Without the `param`, the argument isn't documented like I want it to be.

Screen Shot 2020-01-27 at 3 20 20 PM

**Pull request merged**